File: RemoveRecursionRefactoring/tests/ro/pub/cs/diploma/ExtractRecursiveCallsToStatementsTest.java
package ro.pub.cs.diploma;

import ro.pub.cs.diploma.inspections.ExtractRecursiveCallsToStatementsInspection;

public class ExtractRecursiveCallsToStatementsTest extends IGQuickFixesTestCase {
  @Override
  protected void setUp() throws Exception {
    super.setUp();
    myFixture.enableInspections(new ExtractRecursiveCallsToStatementsInspection());
    myRelativePath = "extract-recursive-calls-to-statements";
    myDefaultHint = RemoveRecursionBundle.message("extract.recursive.calls.to.statements");
  }

  public void testExtract1() {
    doTest();
  }

  public void testExtractForCondition() {
    doTest();
  }

  public void testExtractForEachValue() {
    doTest();
  }

  public void testExtractForInitializer1() {
    doTest();
  }

  public void testExtractForInitializer2() {
    doTest();
  }

  public void testExtractForUpdate() {
    doTest();
  }

  public void testExtractIfCondition() {
    doTest();
  }

  public void testExtractNested() {
    doTest();
  }

  public void testExtractNoChange1() {
    doTest();
  }

  public void testExtractNoChange2() {
    doTest();
  }

  public void testExtractSwitchCondition() {
    doTest();
  }

  public void testExtractWhileCondition() {
    doTest();
  }
}

File: RemoveRecursionRefactoring/tests/ro/pub/cs/diploma/ExtractRecursiveCallsToStatementsTest.kt
package ro.pub.cs.diploma

import ro.pub.cs.diploma.inspections.ExtractRecursiveCallsToStatementsInspection

class ExtractRecursiveCallsToStatementsTest : IGQuickFixesTestCase() {
  @Throws(Exception::class)
  override fun setUp() {
    super.setUp()
    myFixture.enableInspections(ExtractRecursiveCallsToStatementsInspection())
    myRelativePath = "extract-recursive-calls-to-statements"
    myDefaultHint = RemoveRecursionBundle.message("extract.recursive.calls.to.statements")
  }

  fun testExtract1() {
    doTest()
  }

  fun testExtractForCondition() {
    doTest()
  }

  fun testExtractForEachValue() {
    doTest()
  }

  fun testExtractForInitializer1() {
    doTest()
  }

  fun testExtractForInitializer2() {
    doTest()
  }

  fun testExtractForUpdate() {
    doTest()
  }

  fun testExtractIfCondition() {
    doTest()
  }

  fun testExtractNested() {
    doTest()
  }

  fun testExtractNoChange1() {
    doTest()
  }

  fun testExtractNoChange2() {
    doTest()
  }

  fun testExtractSwitchCondition() {
    doTest()
  }

  fun testExtractWhileCondition() {
    doTest()
  }
}
